src/user.rs

```diff
@@ -19,10 +20,22 @@ use utils;
 /// 2. Add the user's name and public key to the protonfile
 ///
 /// Impure.
-pub fn new_user<P: AsRef<Path>>(public_key_path: P, name: &str) -> Result<(), Error> {
+pub fn new_user<P: AsRef<Path>>(
+    admin_key_path: P,
+    public_key_path: P,
+    name: &str
+) -> Result<(), Error> {
+
+    // See if admin has permission to add user
+    let admin_user = try!(id_user(admin_key_path));
```

Comment:
You use this code in new_user, new_sequence, and maybe elsewhere? I think you can pull it out into something like `fn validate_administrator(admin_key_path: &str) -> Result<User, Error>`.